quit/basicfunction.py

```diff
@@ -0,0 +1,138 @@
+import typing
+
+import numpy as np
+from scipy import linalg
+
+Array1D = typing.Union[np.ndarray, typing.Sequence[complex]]
+
+
+def ket(psi: Array1D) -> np.ndarray:
+    """Create a ket |psi> as a vertical vector.
+
+    :param psi: Sequence of complex numbers.
+    :return: A vertical vector as numpy array.
+    """
+    psi = np.asarray(psi)
+
+    if len(psi.squeeze().shape) != 1:
+        raise ValueError("Wrong parameter psi.")
+    return psi.reshape((-1, 1))
+
+
+def basis(dim: int, index: int) -> np.ndarray:
+    """Create computational basis of dimension dim.
+
+    :param dim: Dimension of vector space.
+    :param index:
+    :return: A basis vector |d, i>.
+    """
+    if index >= dim:
+        raise Exception("Parameter index out of range.")
+
+    vec = np.zeros(dim)
+    vec[index] = 1
+    return vec
+
+
+def dagger(matrix: np.ndarray) -> np.ndarray:
+    """Create conjugate transpose of a given matrix.
+
+    :param matrix: A matrix given by numpy array.
+    :return: An operator dagger as concatanation conjugate and transpose.
+    """
+    return np.transpose(np.conjugate(matrix))
+
+
+def bra(psi: Array1D) -> np.ndarray:
+    """Create bra <psi| as a horizontal vector.
+
+    :param psi: Sequence of complex numbers.
+    :return: A horizontal vector as numpy array satisfying <psi| = |psi>^+
+    """
+    return dagger(ket(psi))
+
+
+def braket(psi: Array1D, phi: Array1D) -> complex:
+    """Create braket as an inner product between vectors psi and phi.
+
+    :param psi: Sequence of complex numbers.
+    :param phi: Sequence of complex numbers.
+    :return: Inner product between vectors psi and phi.
+    """
+    return complex(bra(psi) @ ket(phi))
+
+
+def ketbra(psi: Array1D, phi: Array1D) -> np.ndarray:
+    """Create a matrix from multiplication vectors psi and phi.
```

Comment:
It seems to me that this sentence is incorrect, both on language and semantic levels. Maybe we should use Dirac notation directly instead?

```suggestion
    """Compute outer product |psi><phi|.
```